core/json/src/main/scala/net/liftweb/json/JsonAST.scala

```diff
@@ -295,48 +432,65 @@ object JsonAST {
       find(this)
     }
 
-    /** Return the first element from JSON which matches the given predicate.
-      * <p>
-      * Example:<pre>
-      * JArray(JInt(1) :: JInt(2) :: Nil) find { _ == JInt(2) } == Some(JInt(2))
-      * </pre>
-      */
+    /**
+     * Return the first element from a `JValue` which matches the given predicate.
+     *
+     * Example:
+     *
+     * {{{
+     * JArray(JInt(1) :: JInt(2) :: Nil) find { _ == JInt(2) } == Some(JInt(2))
+     * }}}
+     */
     def find(p: JValue => Boolean): Option[JValue] = {
       def find(json: JValue): Option[JValue] = {
-        if (p(json)) return Some(json)
         json match {
+          case _ if p(json) => Some(json)
           case JObject(fs) => fs.flatMap { case JField(n, v) => find(v) }.headOption
           case JArray(l) => l.flatMap(find _).headOption
           case _ => None
         }
       }
+
       find(this)
     }
 
-    /** Return a List of all fields that match the given predicate.
-     * <p>
-     * Example:<pre>
+    /**
+     * Return a List of all fields that match the given predicate.
```

Comment:
May be worth explicitly saying this doesn't recurse into child objects and such?